examples/virtual_methods/cat.rs

```diff
@@ -96,7 +96,17 @@ impl Default for Cat {
 ///
 /// By convention we still create an empty `CatImpl` trait, this allows us to add
 /// 'protected' cat methods only available to be called by other Cats later.
-pub trait CatImpl: PetImpl {}
+pub trait CatImpl: PetImpl
+where
+    <Self as ObjectSubclass>::Type: IsA<Pet>,
```

Comment:
This one is not inferred by requiring `PetImpl`, and `PetImpl` requiring `IsA<Pet>`?